extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -229,6 +235,45 @@ public void setPostLogoutPath(Optional<String> postLogoutPath) {
         }
     }
 
+    /**
+     * Default session manager configuration
+     */
+    @ConfigGroup
+    public static class SessionManager {
+
+        /**
+         * Default SessionManager will only keep ID token out of three tokens (ID, access and refresh)
+         * returned in the authorization code grant response if this property is enabled.
+         */
+        @ConfigItem(defaultValue = "false")
+        public boolean keepIdTokenOnly;
```

Comment:
Maybe an `enum` such as `CookieStrategy` which `SINGLE`, `SPLIT`, `IDTOKEN`, `ACCESSTOKEN`, `REFRESHTOKEN`.

Where people could choose what type of token should be included in the token. For instance, `ACCESSTOKEN` might be useful if the application wants to relay the token to other services while still tracking session lifetime by either a local introspection or a call to the introspection endpoint.

Reply:
@pedroigor I'm really not sure about `SINGLE, SPLIT, IDTOKEN, ACCESSTOKEN, REFRESHTOKEN`. One thing that we always need to keep is the ID token. AT and RT are optional. ID token is what represents the user authentication so we can't ever drop it. 
What would `ACCESSTOKEN` mean - keep ID token and ACCESSTOKEN ? You said the `local introspection` but this is not ID token, they are not really the same thing... We've also had the user issues to do with the binary access tokens not remotely introspectable due to Azure, Google, etc not having an introspection endpoint - you may recall the users started saying their Quarkus apps stopped working after I updated OIDC to verify the code flow AT tokens by default :-)
 
How about this though, I will introduce the enum instead but keep ID token only there - this way we will be able to add more options like ID and access token (combined), etc...
`splitTokens` should still remain a boolean though I think - as it will work with 1 or more tokens combined